awswrangler/distributed/ray/modin/_core.py

```diff
@@ -1,12 +1,41 @@
 """Modin on Ray Core module (PRIVATE)."""
 # pylint: disable=import-outside-toplevel
+import logging
 from functools import wraps
-from typing import Any, Callable, Optional
+from typing import Any, Callable, List, Optional
 
 import pandas as pd
+import ray
 from modin.distributed.dataframe.pandas import from_partitions, unwrap_partitions
 from modin.pandas import DataFrame as ModinDataFrame
 
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+def _validate_partition_cols(df: pd.DataFrame) -> bool:
+    """
+    Validate if partitions of the data frame are partitioned along row axis.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Modin data frame
+
+    Returns
+    -------
+    bool
+    """
+
+    @ray.remote
+    def _validate_partition(df: pd.DataFrame, n_columns: int) -> bool:
+        return len(df.columns) == n_columns
+
+    # Unwrap partitions as they are currently stored (axis=None)
+    partitions: List[List[ray.types.ObjectRef[pd.DataFrame]]] = unwrap_partitions(df, axis=None)
+    return all(
+        ray.get([_validate_partition.remote(partition, len(df.columns)) for row in partitions for partition in row])
```

Comment:
Can you add a comment explaining why `partitions` is a **list of lists** of block references? I'm confused by this.

Also, `partitions` contains `row`, and `row` contains `partition`? Is there any alternate naming we could use here? 😅 When I see `row` I think of row in a DataFrame, and I don't think that's what's being checked here.

Reply:
Haha fair, it's a list of lists because of how Modin [partitions](https://modin.readthedocs.io/en/latest/usage_guide/optimization_notes/index.html#understanding-modin-s-partitioning-mechanism) the data frame - basically it's a matrix of blocks. I will clarify this, thanks.

Actually looking at failed tests I suspect might not always be list of lists.